File: Coders/DSA Concepts/stack.c
#include <stdio.h>

int MAX = 10;

int a[10]; // Maximum size of Stack
int top=-1;

void push(int x)
{
    if (top >= (MAX - 1))
    {
        printf("Stack Overflow");
    }
    else
    {
        a[++top] = x;
        printf("%d  pushed into stack\n",x);
        
    }
}

int pop()
{
    if (top < 0)
    {
        printf("Stack Underflow");
        return 0;
    }
    else
    {
        int x = a[top--];
        return x;
    }
}

int peek()
{
    if (top < 0)
    {
        printf("Stack Underflow");
        return 0;
    }
    else
    {
        int x = a[top];
        return x;
    }
}

void print()
{
    for (int i = top; i > -1; i--)
    {
        printf("%d ", a[i]);
    }
}


int main()
{

        push(1);
        push(2);
        push(3);
        push(4);
        push(5);
        printf("%d Popped from stack\n",pop());
        printf("Top element is : %d\n", peek());
        printf("Elements present in stack :\n");
        print();
}

File: Coders/DSA Concepts/Stack.java
public class Stack
{
    private int arr[];
    private int top;
    private int capacity;
    Stack(int size)
    {
        arr = new int[size];
        capacity = size;
        top = -1;
    }
    public void push(int x)
    {
        if (isFull())
        {
            System.out.println("Overflow\n");
            System.exit(-1);
        }

        System.out.println("Inserting " + x);
        arr[++top] = x;
    }
    public int pop()
    {
        if (isEmpty())
        {
            System.out.println("Underflow\n");
            System.exit(-1);
        }
        System.out.println("Removing " + peek());
        return arr[top--];
    }
    public int peek()
    {
        if (!isEmpty()) {
            return arr[top];
        }
        else {
            System.exit(-1);
        }

        return -1;
    }

    public int size() {
        return top + 1;
    }

    public boolean isEmpty() {
        return top == -1;
    }
    public boolean isFull() {
        return top == capacity - 1;
    }

    public static void main (String[] args)
    {
        Stack stack = new Stack(3);
        stack.push(3);
        stack.push(2);
        stack.push(5);
        stack.pop();
        stack.pop();
        stack.push(3);
        System.out.println("The top element is " + stack.peek());
    }
}
